### PR TITLE
[DNM] Merge Windows Explorations Into Development

### PR DESCRIPTION
**Pull request by brianmichel**
Let's get these branches in line and make sure we have one shared branch to use

### CODE REVIEW

File: Sources/ComposableArchitecture/Internal/Create.swift [hunk 1]
```diff
@@ -20,25 +23,25 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import Combine
-import Darwin
+// import Darwin
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Internal/Create.swift]: This needs fixing

File: Sources/ComposableArchitecture/Internal/Create.swift [hunk 2]
```diff
@@ -20,25 +23,25 @@
...
 
 final class DemandBuffer<S: Subscriber>: @unchecked Sendable {
   private var buffer = [S.Input]()
   private let subscriber: S
   private var completion: Subscribers.Completion<S.Failure>?
   private var demandState = Demand()
-  private let lock: os_unfair_lock_t
+  private let lock: NSRecursiveLock
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Internal/Create.swift]: This should probably switch off lock types depending on whether or not you can link Darwain

File: Sources/ComposableArchitecture/Internal/Create.swift [hunk 3]
```diff
@@ -20,25 +23,25 @@
...
 
   init(subscriber: S) {
     self.subscriber = subscriber
-    self.lock = os_unfair_lock_t.allocate(capacity: 1)
-    self.lock.initialize(to: os_unfair_lock())
+    self.lock = NSRecursiveLock()
+    // self.lock = os_unfair_lock_t.allocate(capacity: 1)
+    // self.lock.initialize(to: os_unfair_lock())
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Internal/Create.swift]: Same thoughts here as above

File: Package.swift [hunk 4]
```diff
@@ -18,12 +18,16 @@ let package = Package(
   ],
   dependencies: [
     .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
+    .package(url: "https://github.com/OpenCombine/OpenCombine.git", from: "0.13.0"),
```
Comment (brianmichel) [inline Package.swift:21]: We likely want to ensure this is only pulled in on the Windows platform.

File: Package.swift [hunk 5]
```diff
@@ -18,12 +18,16 @@ let package = Package(
...
     .package(url: "https://github.com/google/swift-benchmark", from: "0.1.0"),
-    .package(url: "https://github.com/pointfreeco/combine-schedulers", from: "0.8.0"),
+    // currently produces a warning in SPM, but overrrides `Depdendencies`'s version with a version
+    // that supports Windows via OpenCombine(Shims) (to fix warning, see thebrowsercompany/swift-dependencies below)
+    .package(url: "https://github.com/thebrowsercompany/combine-schedulers", branch: "feature/windows-explorations"),
```
Comment (brianmichel) [inline Package.swift]: TODO

File: Package.swift [hunk 6]
```diff
@@ -18,12 +18,16 @@ let package = Package(
...
     .package(url: "https://github.com/pointfreeco/swift-case-paths", from: "0.14.0"),
     .package(url: "https://github.com/apple/swift-collections", from: "1.0.2"),
     .package(url: "https://github.com/pointfreeco/swift-custom-dump", from: "0.9.1"),
-    .package(url: "https://github.com/pointfreeco/swift-dependencies", from: "0.2.0"),
+    // here to also link to https://github.com/thebrowsercompany/combine-schedulers, which supports OpenCombine
+    .package(url: "https://github.com/thebrowsercompany/swift-dependencies", branch: "feature/windows-explorations"),
```
Comment (brianmichel) [inline Package.swift]: TODO

File: Package.swift [hunk 7]
```diff
@@ -32,15 +36,19 @@ let package = Package(
     .target(
       name: "ComposableArchitecture",
       dependencies: [
+        .product(name: "OpenCombineShim", package: "OpenCombine"),
```
Comment (brianmichel) [inline Package.swift:39]: This should only be added for Windows builds

File: Package.swift [hunk 8]
```diff
@@ -60,6 +68,23 @@ let package = Package(
   ]
 )
 
+func osSpecificComposableArchitectureExcludes() -> [String] {
```
Comment (brianmichel) [inline Package.swift:71]: Needs documentation

File: Sources/ComposableArchitecture/Effect.swift [hunk 9]
```diff
@@ -1,8 +1,11 @@
-import Combine
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effect.swift:1]: This should be wrapped in an canImport check

File: Sources/ComposableArchitecture/Effects/Cancellation.swift [hunk 10]
```diff
@@ -1,5 +1,6 @@
-import Combine
 import Foundation
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Cancellation.swift:2]: Should be wrapped in a canImport check

File: Sources/ComposableArchitecture/Effects/Publisher.swift [hunk 11]
```diff
@@ -1,17 +1,4 @@
...
-  /// - Parameter createPublisher: The closure to execute when the effect is performed.
-  /// - Returns: An effect wrapping a Combine publisher.
-  public static func publisher<P: Publisher>(_ createPublisher: @escaping () -> P) -> Self
-  where P.Output == Action, P.Failure == Never {
-    Self(
-      operation: .publisher(Deferred(createPublisher: createPublisher).eraseToAnyPublisher())
-    )
-  }
-}
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Publisher.swift:1]: Should be wrapped in an canImport check

File: Sources/ComposableArchitecture/Effects/Publisher/Debouncing.swift [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import Combine
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Publisher/Debouncing.swift:1]: Should be wrapped with a canImport check

File: Sources/ComposableArchitecture/Effects/Publisher/Deferring.swift [hunk 13]
```diff
@@ -1,4 +1,4 @@
-import Combine
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Publisher/Deferring.swift:1]: Should be wrapped with a canImport check

File: Sources/ComposableArchitecture/Effects/Publisher/Throttling.swift [hunk 14]
```diff
@@ -1,6 +1,6 @@
-import Combine
 import Dispatch
 import Foundation
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Publisher/Throttling.swift:3]: Should be wrapped with a canImport check

File: Sources/ComposableArchitecture/Effects/Publisher/Timer.swift [hunk 15]
```diff
@@ -1,5 +1,5 @@
-import Combine
 import CombineSchedulers
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Effects/Publisher/Timer.swift:2]: Should be wrapped with a canImport check

File: Sources/ComposableArchitecture/Internal/CurrentValueRelay.swift [hunk 16]
```diff
@@ -1,5 +1,5 @@
-import Combine
 import Foundation
+import OpenCombineShim
```
Comment (brianmichel) [inline Sources/ComposableArchitecture/Internal/CurrentValueRelay.swift:2]: Should be wrapped with a canImport check

File: Sources/ComposableArchitecture/Reducer/AnyReducer/AnyReducer.swift [hunk 17]
```diff
@@ -1,5 +1,5 @@
 import CasePaths
-import Combine
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/Reducer/AnyReducer/AnyReducer.swift:2]: Does this need to be wrapped?

File: Sources/ComposableArchitecture/Reducer/AnyReducer/AnyReducerSignpost.swift [hunk 18]
```diff
@@ -1,152 +1,154 @@
-import Combine
-import os.signpost
+#if canImport(OSLog)
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/Reducer/AnyReducer/AnyReducerSignpost.swift:1]: Why `OSLog`?

File: Sources/ComposableArchitecture/Store.swift [hunk 19]
```diff
@@ -1,5 +1,5 @@
-import Combine
 import Foundation
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/Store.swift:2]: Ditto wrapping?

File: Sources/ComposableArchitecture/TestStore.swift [hunk 20]
```diff
@@ -1,5 +1,5 @@
 @_spi(Internals) import CasePaths
-import Combine
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/TestStore.swift:2]: Ditto

File: Sources/ComposableArchitecture/TestStore.swift [hunk 21]
```diff
@@ -2433,7 +2433,9 @@ private func _XCTExpectFailure(
   strict: Bool = true,
   failingBlock: () -> Void
 ) {
+
   #if DEBUG
+  #if os(macOS)
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/TestStore.swift:2438]: Is this ours or upstream? If ours, why `macOS` and not `Darwin`? When do we decide?

File: Sources/ComposableArchitecture/UIKit/IfLetUIKit.swift [hunk 22]
```diff
@@ -1,4 +1,4 @@
-import Combine
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/UIKit/IfLetUIKit.swift:1]: Ditto

File: Sources/ComposableArchitecture/ViewStore.swift [hunk 23]
```diff
@@ -1,5 +1,8 @@
-import Combine
-import SwiftUI
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/ComposableArchitecture/ViewStore.swift:1]: Ditto

File: Sources/swift-composable-architecture-benchmark/Dependencies.swift [hunk 24]
```diff
@@ -1,5 +1,5 @@
 import Benchmark
-import Combine
+import OpenCombineShim
```
Comment (amonshiz) [inline Sources/swift-composable-architecture-benchmark/Dependencies.swift:2]: Ditto

File: Tests/ComposableArchitectureTests/Hacks.swift [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+private func warnFail(message: String) -> String {
+  return """
+    XCTExpectFailure: \(message)
+
+    ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┉┅
+    ┃ ⚠︎ Warning: This XCTExpectFailure was ignored.
+    ┃
+    ┃ XCTExpectFailure is currently not supported on Windows.
+    ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┉┅
+        ▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄▀▄
```
Comment (amonshiz) [inline Tests/ComposableArchitectureTests/Hacks.swift:49]: Fancy

**Review by amonshiz — COMMENTED** [review]
Just a drive-by sorry

**Comment by brianmichel** [comment]
Closing in favor ov https://github.com/thebrowsercompany/swift-composable-architecture/pull/49

**Pull request closed without merging**